File: src/main/java/org/code4everything/hutool/converter/SetStringConverter.java
package org.code4everything.hutool.converter;

import org.code4everything.hutool.Converter;

import java.util.HashSet;
import java.util.Set;

/**
 * @author pantao
 * @since 2021/5/11
 */
public class SetStringConverter implements Converter<Set<String>> {

    private final ListStringConverter converter = new ListStringConverter();

    @Override
    public Set<String> string2Object(String string) {
        return new HashSet<>(converter.string2Object(string));
    }

    @Override
    @SuppressWarnings("unchecked")
    public String object2String(Object object) throws Exception {
        return object instanceof Set ? converter.object2String(object) : "";
    }
}

File: src/main/kotlin/org/code4everything/hutool/converter/SetStringConverter.kt
package org.code4everything.hutool.converter

import org.code4everything.hutool.Converter

class SetStringConverter : Converter<Set<String?>> {

    private val converter = ListStringConverter()

    override fun string2Object(string: String): Set<String> = HashSet(converter.string2Object(string))

    override fun object2String(any: Any): String = if (any is Set<*>) converter.object2String(any) else ""
}
